src/subplots.ts

```diff
@@ -2939,6 +2975,83 @@ export class newParallelPlot extends Figure {
+    public Y: number,
+    public canvas_id: string,
+    public is_in_multiplot: boolean = false
+    ) {
+      super(data, width, height, buttons_ON, X, Y, canvas_id, is_in_multiplot);
+      this.relativeObjects = ShapeCollection.fromPrimitives(data.primitives, this.scale);
+    }
+
+  public shiftOnAction(canvas: HTMLElement): void {}
+
+  public define_canvas(canvas_id: string):void {
+    super.define_canvas(canvas_id);
+    this.computeTextBorders(this.context_show);
+  }
+
+  public reset_scales(): void { // TODO: merge with resetView
+    super.reset_scales();
+    this.updateBounds();
+  }
+
+  protected unpackData(data: any): Map<string, any[]> {
+    const drawing = ShapeCollection.fromPrimitives(data.primitives);
+    const [minX, minY, maxX, maxY] = Draw.boundsDilatation(...drawing.getBounds());
+    return new Map<string, any[]>([["x", [minX, maxX]], ["y", [minY, maxY]], ["shapes", drawing.shapes]])
+  }
+
+  private static boundsDilatation(minimum: Vertex, maximum: Vertex): [number, number, number, number] {
+    const minX = minimum.x * (1 - Math.sign(minimum.x) * DRAW_MARGIN_FACTOR);
+    const minY = minimum.y * (1 - Math.sign(minimum.y) * DRAW_MARGIN_FACTOR);
+    const maxX = maximum.x * (1 + Math.sign(maximum.x) * DRAW_MARGIN_FACTOR);
+    const maxY = maximum.y * (1 + Math.sign(maximum.y) * DRAW_MARGIN_FACTOR);
+    return [minX, minY, maxX, maxY]
+  }
+
+  protected computeTextBorders(context: CanvasRenderingContext2D) {
+    this.relativeObjects.updateBounds(context);
+    this.updateBounds();
+  }
+
+  public updateBounds(): void {
+    const [minX, minY, maxX, maxY] = Draw.boundsDilatation(this.relativeObjects.minimum, this.relativeObjects.maximum);
+    this.axes[0].minValue = this.features.get("x")[0] = Math.min(this.features.get("x")[0], minX);
+    this.axes[1].minValue = this.features.get("y")[0] = Math.min(this.features.get("y")[0], minY);
+    this.axes[0].maxValue = this.features.get("x")[1] = Math.max(this.features.get("x")[1], maxX);
+    this.axes[1].maxValue = this.features.get("y")[1] = Math.max(this.features.get("y")[1], maxY);
+    this.axes.forEach(axis => axis.saveLocation());
+    this.axisEqual();
+  }
+
+  protected drawRelativeObjects(context: CanvasRenderingContext2D) { this.drawInZone(context) }
+
+  protected updateDrawnObjects(context: CanvasRenderingContext2D): void {
+    this.relativeObjects.locateLabels(super.cuttingZone, this.initScale);
+    this.relativeObjects.draw(context);
+  }
+
+  protected get cuttingZone(): newRect {
+    const axesOrigin = this.axes[0].origin.transform(this.frameMatrix.inverse());
+    const axesEnd = new Vertex(this.axes[0].end.x, this.axes[1].end.y).transform(this.frameMatrix.inverse());
+    return new newRect(axesOrigin, axesEnd.subtract(axesOrigin));
+  }
+
+  protected axisEqual(): void {
+    if (this.axes[0].drawLength / this.axes[0].interval > this.axes[1].drawLength / this.axes[1].interval) this.axes[0].otherAxisScaling(this.axes[1])
```

Comment:
Do you think this condition could be set in a variable to make the intention more obvious and the code more readable ?

Reply:
Set `scale = this.drawLenth / this.interval` as axis property